src/js/libgeo/Prover.js

```diff
@@ -68,21 +74,77 @@ function incidentPC(p, c) {
             var erg = General.mult(c.matrix, p.homog);
             erg = General.mult(p.homog, erg);
             erg = CSNumber.abs(erg);
-            if (erg.value.real < 0.0000000000001) {
-                p.incidences.push(c.name);
-                c.incidences.push(p.name);
-            }
+            return (erg.value.real < 0.0000000000001);
         }
     };
 }
 
 function checkConjectures() {
+    var debug = false;
+    if (debug) console.log("conjectures", conjectures.length);
+    //if (!debug)
     if (conjectures.length === 0) return;
-    // TODO: we need some randomized proving here:
-    // move free elements and check conjectures a number of times.
-    // For now assume that all conjectures could be verified.
+    backupGeo();
```

Comment:
I'd rather rename those to something containing the word *animation*, or move their content into `csplay` and `csstop`.

Having a fixed number of backup states for a variable number of objects works well. Having a variable number of backup states for a fixed number of objects would be fine as well. But since we have `createtool` the number of objects is no longer fixed. And allocating in several places to keep the states in sync is trouble I'd rather avoid unless I see good reason not to. And allocating for each individual backup will likely be too costly. Last I checked, allocating typed arrays was far from cheap.